st_package_reviewer/check/file/__init__.py

```diff
@@ -42,6 +43,18 @@ def file_context(self, path):
             pass
         return self.context("File: {}".format(path))
 
+    def visit_all_pyfiles(self):
+        pyfiles = self.glob("**/*.py")
+        for self.current_file in pyfiles:
+            with open(self.current_file, "r") as f:
+                try:
+                    root = ast.parse(f.read(), self.current_file)
```

Comment:
We can improve runtime by caching the generated AST (and also the `pyfiles` variable) through a `functools.lrucache`-wrapped classmethod. This will also only raise one error for a file that fails to parse instead of one for checker.